src/librustc_builtin_macros/format.rs

```diff
@@ -1029,6 +1029,20 @@ pub fn expand_preparsed_format_args(
         }
         e.emit();
         return DummyResult::raw_expr(sp, true);
+    } else if !parser.warnings.is_empty() {
+        let warn = parser.warnings.remove(0);
+        let sp = fmt_span.from_inner(warn.span);
+        let mut w =
+            ecx.struct_span_warn(sp, &format!("format string warning: {}", warn.description));
```

Comment:
This should be emitted as a lint rather than a warning using `ecx.parse_sess.buffer_lint_with_diagnostic(...)`.

Reply:
cool, thanks :)